skyportal/handlers/api/photometry.py

```diff
@@ -337,18 +338,104 @@ def post(self):
+            if len(df_existing_photometry) > 0:
+                df_existing_photometry["hash"] = df_existing_photometry.apply(
+                    lambda row: hash(
+                        (
+                            row["instrument_id"],
+                            row["filter"],
+                            row["mjd"],
+                            row["flux"],
+                            row["fluxerr"],
+                            row["origin"],
+                        )
+                    ),
+                    axis=1,
+                )
+                df_existing_photometry["group_ids"] = df_existing_photometry.apply(
+                    lambda row: set([g.id for g in row["groups"]]), axis=1
+                )
+
+                w_oid_duplicate = (df.obj_id == oid) & (
+                    df.hash.isin(df_existing_photometry.hash)
+                )
+
+                if any(w_oid_duplicate):
+                    # update photometry.groups if necessary
+                    for row in df.loc[w_oid_duplicate].itertuples():
+                        # select the corresponding row in existing photometry by the pre-computed hash:
+                        w = df_existing_photometry.hash == row.hash
+                        # select the photometry point's group_ids:
+                        row_group_ids = set(
+                            df_existing_photometry.loc[w, "group_ids"].values[0]
+                        )
+                        # posting to new groups?
+                        if len(set(group_ids) - row_group_ids) > 0:
+                            # select old + new groups
+                            group_ids_update = set(group_ids).union(row_group_ids)
+                            groups = (
+                                DBSession()
+                                .query(Group)
+                                .filter(Group.id.in_(group_ids_update))
+                                .all()
+                            )
+                            # update the corresponding photometry entry in the db
+                            index_to_update = np.where(w)[0][0]
+                            existing_photometry[index_to_update].groups = groups
+
+                            updated_photometry_groups = True
+
+                    # now safely drop the duplicates:
+                    df = df.drop(index=np.where(w_oid_duplicate)[0])
+
+        # posted data contains only duplicates already present in the db?
+        if len(df) == 0:
+            if updated_photometry_groups:
+                DBSession().commit()
+                return self.success(
+                    data={
+                        "ids": [],
+                        "message": "Photometry already exists, updated photometry groups.",
+                    }
+                )
+            else:
+                return self.success(
+                    data={"ids": [], "message": "Photometry already exists."}
+                )
```

Comment:
@stefanv @danny if it so happens that after all the checks no new photometry gets posted bc it's already in the db, should I return "error" or "success" with ids=[] as I do here?

Reply:
I would return 2 keys: ids of existing points, (`ids`) and ids of new points (if any) (`new_ids`)

Reply:
Oh boy, one more thing to keep track of... This handler has grown into a beast, impenetrable to an external observer!

Reply:
Just whining :)